### PR TITLE
fix(overlay): fix typo in property name

### LINKED ISSUE #12925: Typo in cdk/overlay/overlay-directives.ts
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Public property should be "flexibleDimensions"

#### What is the current behavior?
Public property is "flexibleDiemsions"

#### What are the steps to reproduce?
https://github.com/angular/material2/blob/master/src/cdk/overlay/overlay-directives.ts

Lines 186 and 187

#### What is the use-case or motivation for changing an existing behavior?
Public-facing property is incorrectly presented, causing confusion

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Latest

### PR DESCRIPTION
**Pull request by CaerusKaru**
BREAKING CHANGE:
* the `flexibleDiemsions` on `CdkConnectedOverlay` has been renamed to `flexibleDimensions`

Fixes #12925

### CODE REVIEW

**Comment by jelbourn** [comment]
We'll need an `ng update` fix for this one; @CaerusKaru want to try making one?

**Comment by CaerusKaru** [comment]
@jelbourn Sure, I can do that later today.

**Comment by CaerusKaru** [comment]
@jelbourn should be all set

File: src/lib/schematics/update/material/data/property-names.ts [hunk 1]
```diff
@@ -335,5 +335,18 @@ export const propertyNames = transformChanges<MaterialPropertyNameData>([
         }
       }
     ]
+  },
+
+  {
```
Comment (DevVersion) [inline src/lib/schematics/update/material/data/property-names.ts]: Please add a comment `// TODO(devversion): this should be part of the V6 to V7 upgrade`.

**Comment by jelbourn** [comment]
@crisbeto technically.................yeah. But nobody is using it in Google, and since it's such a blatant typo, I was going to let this one slide.

**Comment by CaerusKaru** [comment]
@jelbourn et al sorry for the commit spam. I momentarily lost my mind trying to remove a modified package-lock from the commit. I was waiting on @jelbourn's sign-off re @crisbeto's point before adding @DevVersion's comment. This PR should be good to go :+1:

**Pull request merged**